pkg/agent/cniserver/server_linux_test.go

```diff
@@ -0,0 +1,742 @@
+		ipamMock.EXPECT().Check(gomock.Any(), gomock.Any(), gomock.Any()).Return(true, nil).Times(1)
+		resp, err := cniserver.CmdCheck(ctx, requestMsg)
+		assert.NoError(t, err)
+		assert.Equal(t, emptyResponse, resp)
+	})
+}
+
+func TestReconcile(t *testing.T) {
+	controller := gomock.NewController(t)
+	defer controller.Finish()
+	mockOVSBridgeClient = ovsconfigtest.NewMockOVSBridgeClient(controller)
+	mockOFClient = openflowtest.NewMockClient(controller)
+	ifaceStore = interfacestore.NewInterfaceStore()
+	routeMock = routetest.NewMockInterface(controller)
+	nodeName := "node1"
+	cniServer := newCNIServer(t)
+	cniServer.routeClient = routeMock
+	gwMAC, _ := net.ParseMAC("00:00:11:11:11:11")
+	cniServer.podConfigurator, _ = newPodConfigurator(mockOVSBridgeClient, mockOFClient, routeMock, ifaceStore, gwMAC, "system", false, channel.NewSubscribableChannel("PodUpdate", 100), nil, false)
+	cniServer.podConfigurator.ifConfigurator = newTestInterfaceConfigurator()
+	cniServer.nodeConfig = &config.NodeConfig{
+		Name: nodeName,
+	}
+	pods := []*v1.Pod{
+		{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "p1",
+				Namespace: testPodNamespace,
+			},
+			Spec: v1.PodSpec{
+				NodeName: nodeName,
+			},
+		},
+		{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "p2",
+				Namespace: testPodNamespace,
+			},
+			Spec: v1.PodSpec{
+				NodeName:    nodeName,
+				HostNetwork: true,
+			},
+		},
+		{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "p4",
+				Namespace: testPodNamespace,
+			},
+			Spec: v1.PodSpec{
+				NodeName: nodeName,
+			},
+		},
+	}
+	containerIfaces := map[string]*interfacestore.InterfaceConfig{
+		"iface1": {
+			InterfaceName: "iface1",
+			Type:          interfacestore.ContainerInterface,
+			OVSPortConfig: &interfacestore.OVSPortConfig{
+				PortUUID: generateUUID(t),
+				OFPort:   int32(3),
+			},
+			ContainerInterfaceConfig: &interfacestore.ContainerInterfaceConfig{
+				PodName:      "p1",
+				PodNamespace: testPodNamespace,
+				ContainerID:  generateUUID(t),
+			},
+		},
+		"iface3": {
+			InterfaceName: "iface3",
+			Type:          interfacestore.ContainerInterface,
+			OVSPortConfig: &interfacestore.OVSPortConfig{
+				PortUUID: generateUUID(t),
+				OFPort:   int32(4),
+			},
+			ContainerInterfaceConfig: &interfacestore.ContainerInterfaceConfig{
+				PodName:      "p3",
+				PodNamespace: testPodNamespace,
+				ContainerID:  generateUUID(t),
+			},
+		},
+		"iface4": {
+			InterfaceName: "iface4",
+			Type:          interfacestore.ContainerInterface,
+			OVSPortConfig: &interfacestore.OVSPortConfig{
+				PortUUID: generateUUID(t),
+				OFPort:   int32(-1),
+			},
+			ContainerInterfaceConfig: &interfacestore.ContainerInterfaceConfig{
+				PodName:      "p4",
+				PodNamespace: testPodNamespace,
+				ContainerID:  generateUUID(t),
+			},
+		},
+	}
+	objects := make([]runtime.Object, 0)
+	for _, p := range pods {
+		objects = append(objects, runtime.Object(p))
+	}
```

Comment:
You can define the pods the same as you did in window test file without this loop.